core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -137,6 +137,15 @@ class DAGScheduler(
   private[scheduler] val eventProcessLoop = new DAGSchedulerEventProcessLoop(this)
   taskScheduler.setDAGScheduler(this)
 
+  // Number of map, reduce tasks above which we do not assign preferred locations
+  // based on map output sizes. We limit the size of jobs for which assign preferred locations
+  // as sorting the locations by size becomes expensive.
+  private[this] val SHUFFLE_PREF_MAP_THRESHOLD = 1000
+  // NOTE: This should be less than 2000 as we use HighlyCompressedMapStatus beyond that
+  private[this] val SHUFFLE_PREF_REDUCE_THRESHOLD = 1000
+  // Number of preferred locations to use for reducer tasks
+  private[scheduler] val NUM_REDUCER_PREF_LOCS = 5
```

Comment:
one more comment request here: can you add one sentence explaining this? something like "making this smaller will focus on the locations where the most data can be read locally, but may lead to lower scheduling efficiency or the delay scheduling timers expiring if all of those locations are busy"